torch/csrc/jit/script/compiler.cpp

```diff
@@ -389,12 +389,17 @@ struct Environment {
     }
 
     if (!retval) {
-      if (auto type = resolver->resolveType(ident)) {
-        if (auto class_type = type->cast<ClassType>()) {
-          retval = std::make_shared<script::ClassValue>(class_type);
-        } else if (auto tuple_type = type->cast<TupleType>()) {
-          retval = std::make_shared<script::NamedTupleConstructor>(tuple_type);
+      try {
```

Comment:
This `try` catches way too many things. What is it suppose to be catching? Generally we avoid all use of try/catch because it messes up debugging.